engine/src/main/java/org/pentaho/di/trans/steps/reservoirsampling/ReservoirSamplingMeta.java

```diff
@@ -0,0 +1,288 @@
+import org.pentaho.di.core.exception.KettleStepException;
+import org.pentaho.di.core.exception.KettleXMLException;
+import org.pentaho.di.core.row.RowMetaInterface;
+import org.pentaho.di.core.variables.VariableSpace;
+import org.pentaho.di.core.xml.XMLHandler;
+import org.pentaho.di.repository.ObjectId;
+import org.pentaho.di.repository.Repository;
+import org.pentaho.di.trans.Trans;
+import org.pentaho.di.trans.TransMeta;
+import org.pentaho.di.trans.step.BaseStepMeta;
+import org.pentaho.di.trans.step.StepDataInterface;
+import org.pentaho.di.trans.step.StepInterface;
+import org.pentaho.di.trans.step.StepMeta;
+import org.pentaho.di.trans.step.StepMetaInterface;
+import org.pentaho.metastore.api.IMetaStore;
+import org.w3c.dom.Node;
+
+/**
+ * Contains the meta data for the ReservoirSampling step.
+ *
+ * @author Mark Hall (mhall{[at]}pentaho.org)
+ * @version 1.0
+ */
+public class ReservoirSamplingMeta extends BaseStepMeta implements StepMetaInterface {
+
+  public static final String XML_TAG = "reservoir_sampling";
+
+  // Size of the sample to output
+  protected String m_sampleSize = "100";
+
+  // Seed for the random number generator
+  protected String m_randomSeed = "1";
+
+  /**
+   * Creates a new <code>ReservoirMeta</code> instance.
+   */
+  public ReservoirSamplingMeta() {
+    super(); // allocate BaseStepMeta
+  }
+
+  /**
+   * Get the sample size to generate.
+   *
+   * @return the sample size
+   */
+  public String getSampleSize() {
+    return m_sampleSize;
+  }
+
+  /**
+   * Set the size of the sample to generate
+   *
+   * @param sampleS
+   *          the size of the sample
+   */
+  public void setSampleSize( String sampleS ) {
+    m_sampleSize = sampleS;
+  }
+
+  /**
+   * Get the random seed
+   *
+   * @return the random seed
+   */
+  public String getSeed() {
+    return m_randomSeed;
+  }
+
+  /**
+   * Set the seed value for the random number generator
+   *
+   * @param seed
+   *          the seed value
+   */
+  public void setSeed( String seed ) {
+    m_randomSeed = seed;
+  }
+
+  /**
+   * Return the XML describing this (configured) step
+   *
+   * @return a <code>String</code> containing the XML
+   */
+  public String getXML() {
+    StringBuilder retval = new StringBuilder( 100 );
+
+    retval.append( XMLHandler.openTag( XML_TAG ) ).append( Const.CR );
+    retval.append( XMLHandler.addTagValue( "sample_size", m_sampleSize ) );
+    retval.append( XMLHandler.addTagValue( "seed", m_randomSeed ) );
+    retval.append( XMLHandler.closeTag( XML_TAG ) ).append( Const.CR );
+
+    return retval.toString();
+  }
+
+  /**
+   * Check for equality
+   *
+   * @param obj
+   *          an <code>Object</code> to compare with
+   * @return true if equal to the supplied object
+   */
+  public boolean equals( Object obj ) {
+    if ( obj != null && ( obj.getClass().equals( this.getClass() ) ) ) {
+      ReservoirSamplingMeta m = (ReservoirSamplingMeta) obj;
+      return ( getXML() == m.getXML() );
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.